app/Imports/FarmSheetImport.php

```diff
@@ -3,60 +3,87 @@
-
-        $location = Location::where('code', $row[$locationCodeColumn])
-            ->where('location_level_id', $locationLevel->id)
-            ->first();
-
-        // Find the identifier columns;
-        $identifierColumns = collect($this->data['farm_identifiers'])->map(fn ($identifier) => $headers[$identifier]);
-
-        // Get the data from those columns;
-        $identifierData = $identifierColumns->mapWithKeys(fn ($column) => [$column => $row[$column]]);
-
-        // Find the property columns;
-        $propertyColumns = collect($this->data['farm_properties'])->map(fn ($property) => $headers[$property]);
-
-        // Get the data from those columns;
-        $propertyData = $propertyColumns->mapWithKeys(fn ($column) => [$column => $row[$column]]);
-
-
-        return new Farm([
-            'owner_id' => $this->data['owner_id'],
-            'owner_type' => $this->data['owner_type'],
-            'location_id' => $location->id,
-            'team_code' => $row[$farmCodeColumn],
-            'identifiers' => $identifierData,
-            'properties' => $propertyData,
-        ]);
+        foreach ($rows as $row) 
+        {
+            $importedFarms = [];
+
+            foreach ($rows as $row) {
+                $headers = $this->data['header_columns'];
+
+                $farmCodeColumn = $headers[$this->data['farm_code_column']];
+
+                $locationLevel = LocationLevel::find($this->data['location_level_id']);
+                $locationCodeColumn = $headers[$this->data['location_code_column']];
+                $location = Location::where('code', $row[$locationCodeColumn])
+                    ->where('location_level_id', $locationLevel->id)
+                    ->first();
+    
+                // Find the identifier columns;
+                $identifierColumns = collect($this->data['farm_identifiers'])->map(fn ($identifier) => $headers[$identifier]);
+                // Get the data from those columns;
+                $identifierData = $identifierColumns->mapWithKeys(fn ($column) => [$column => $row[$column]]);
+    
+                // Find the property columns;
+                $propertyColumns = collect($this->data['farm_properties'])->map(fn ($property) => $headers[$property]);
+                // Get the data from those columns;
+                $propertyData = $propertyColumns->mapWithKeys(fn ($column) => [$column => $row[$column]]);
+    
+                // Create the farm
+                $farm = new Farm([
+                    'owner_id' => $this->data['owner_id'],
+                    'owner_type' => $this->data['owner_type'],
+                    'location_id' => $location->id,
+                    'team_code' => $row[$farmCodeColumn],
+                    'identifiers' => $identifierData,
+                    'properties' => $propertyData,
+                ]);
+                $farm->save();
+
+                // Farm groups
+                foreach ($this->data as $key => $value) {
```

Comment:
This works well and is a great approach assuming we don't know how many groupings there are. Possibly something we can revise in future to loop through the FarmGroupings instead of the `data` values, but not a priority.